Library/Homebrew/cmd/install.rb

```diff
@@ -277,7 +287,21 @@ def run
               require_sha:    args.require_sha?,
               skip_cask_deps: args.skip_cask_deps?,
               verbose:        args.verbose?,
-            ).install
+            )
+            begin
+              installer.install
+            rescue ErrorDuringExecution => e
+              if args.value("non-interactive-sudo").present? && (
+                   e.stderr.include?("a password is required") ||
+                   e.stderr.include?("no tty present") ||
+                   e.message.include?("sudo")
+                 )
+                opoo "Non-interactive mode: sudo/interactive prompt detected in cask #{cask.full_name}. Skipping."
```

Comment:
Does there not need to be an equivalent skip for cask upgrades?

Again: I think this logic probably makes more sense to live inside `Cask::Installer#install` so it's handled appropriately for install/upgrade/reinstall.